docs/tutorials/remote_soc.rst

```diff
@@ -0,0 +1,209 @@
+            // IRQ Interface
+            input      [31:0] irq,
+            output reg [31:0] eoi,
+
+            // Trace Interface
+            output reg        trace_valid,
+            output reg [35:0] trace_data
+    );
+
+        // Memory signals.
+        reg mem_valid, mem_instr, mem_ready;
+        reg [31:0] mem_addr;
+        reg [31:0] mem_wdata;
+        reg [ 3:0] mem_wstrb;
+        reg [31:0] mem_rdata;
+
+        // No 'ready' signal in sky130 SRAM macro; presumably it is single-cycle?
+        always @(posedge clk)
+            mem_ready <= mem_valid;
+
+        // (Signals have the same name as the picorv32 module: use '.*' to autofill)
+        picorv32 rv32_soc (
+          .*
+        );
+
+        // SRAM with always-active chip select and write control bits.
+        sky130_sram_2kbyte_1rw1r_32x512_8 sram (
+            .clk0(clk),
+            .csb0('b0),
+            .web0(!(mem_wstrb != 0)),
+            .wmask0(mem_wstrb),
+            .addr0(mem_addr),
+            .din0(mem_wdata),
+            .dout0(mem_rdata),
+            .clk1(clk),
+            .csb1('b1),
+            .addr1('b0),
+            .dout1()
+        );
+    endmodule
+
+Finally, your core build script will need to be updated to import the new SRAM Library, and specify some extra parameters such as die size and macro placement::
+
+    import siliconcompiler
+
+    design = 'picorv32_top'
+    die_width = 1000
+    die_height = 1000
+
+    chip = siliconcompiler.Chip(design)
+    chip.load_target('skywater130_demo')
+
+    # Set input source files.
+    chip.input(f'{design}.v')
+    chip.input('picorv32.v')
+    chip.input('sky130_sram_2k.bb.v')
+    chip.input('picorv32.sdc')
+
+    # Set die outline and core area.
+    chip.set('constraint', 'outline', [(0,0), (die_w, die_h)])
+    chip.set('constraint', 'corearea', [(10,10), (die_w-10, die_h-10)])
+
+    # Setup SRAM macro library.
+    from sram import sky130_sram_2k
+    chip.use(sky130_sram_2k)
+    chip.add('asic', 'macrolib', 'sky130_sram_2k')
+
+    # SRAM pins are inside the macro boundary; no routing blockage padding is needed.
+    chip.set('tool', 'openroad', 'task', 'route', 'var', 'grt_macro_extension', '0')
+    # Disable CDL file generation until we can find a CDL file for the SRAM block.
+    chip.set('tool', 'openroad', 'task', 'export', 'var', 'write_cdl', 'false')
+
+    # Place macro instance.
+    chip.set('constraint', 'component', 'sram', 'placement', (500.0, 250.0, 0.0))
+    chip.set('constraint', 'component', 'sram', 'rotation', 270)
+
+    # Build on a remote server.
+    chip.set('option', 'remote', True)
+    chip.run()
+
+With all of that done, your top-level build script should take about 20 minutes to run on the cloud servers if they are not too busy, with most of that time spent in the routing task. As with the previous designs, you should see periodic updates on its progress, and you should receive a screenshot and metrics summary once the job is complete.
```

Comment:
I think it would be helpful at this point, to provide what the directory tree looks like, for the user, as a sanity check to make sure all the files got copied over okay and named what the python script is expecting. 

This was my directory tree when I got it to work:
```
<rundir>
├── sram
│ ├── sky130_sram_2k.py
│ ├── sky130_sram_2kbyte_1rw1r_32x512_8.gds
│ └── sky130_sram_2kbyte_1rw1r_32x512_8.lef
├── picorv32.py
├── picorv32.sdc
├── picorv32.v
├── picorv32_top.py
├── picorv32_top.v
└── sky130_sram_2k.bb.v
```

Reply:
Good idea. Maybe we should also have them put the SRAM blackbox file in the `sram/` directory?

Reply:
It would be nicer, but I think that it's okay as is, for now, since this is working and matches with both the code and instructional text.
Maybe a nice-to-have for the next rev, and easier to track/follow encapsulated in its own PR.

Reply:
Oh, whoops - I made the change in the last commit. It was quick because there aren't that many places where a `.bb.v` file extension is used.